Generator/VisitsFromLogs.php

```diff
@@ -70,9 +81,65 @@ public function generate($time = false, $idSite = 1)
         return $count;
     }
 
-    private function getLogFiles()
+    /**
+     * Generate from logs for a specific day of the month within a specific time range.
+     *
+     * @param int $dayOfMonth the day of month of visits to track. log visits outside of this day
+     *                        of the month will not be tracked.
+     * @param int $startTimeOfDay time of day in seconds, start time of log window. logs before this
+     *                            time will not be tracked.
+     * @param int $endTimeOfDay time of day in seconds, end time of log window. logs before this
+     *                          time will not be tracked.
+     * @param int $idSite site to track to
+     * @return int count of visits tracked
+     */
+    public function generateWithinRange($dayOfMonth, $startTimeOfDay, $endTimeOfDay, $idSite)
     {
-        return Filesystem::globr(PIWIK_INCLUDE_PATH . '/plugins/*/data', '*.log');
+        $prefix = $this->getPiwikUrl() . "piwik.php";
+        $languages = Request::getAcceptLanguages();
+        $numLanguages = count($languages);
+
+        $fileIterators = [];
+        $appendIterator = new \AppendIterator();
+        foreach ($this->logFiles as $logFile) {
+            $fileIterator = new FileLineIterator($logFile);
+            $fileIterators[] = $fileIterator;
+
+            $appendIterator->append($fileIterator);
+        }
+
+        $count = 0;
+        try {
+            foreach ($appendIterator as $line) {
+                $log = LogParser::parseLogLine($line);
+                if (empty($log)) {
+                    continue;
+                }
+
+                $logTime = Date::factory($log['time']);
+                $timeOfDay = $logTime->getTimestamp() % 86400;
+
+                if ($this->getDayOfMonthFromTime($logTime) != $dayOfMonth
+                    || $timeOfDay < $startTimeOfDay
+                    || $timeOfDay > $endTimeOfDay
```

Comment:
I haven't looked too much into it, just wondering does one of the comparisons need to have an equal like `>=` or `<=`?